bson/codec_options.py

```diff
@@ -246,36 +246,36 @@ class CodecOptions(Tuple, Generic[_DocumentType]):
         datetime_conversion: Optional[int]
 
         def __new__(
-            cls: Type["CodecOptions"],
+            cls: Type[CodecOptions[_DocumentType]],
             document_class: Optional[Type[_DocumentType]] = ...,
             tz_aware: bool = ...,
             uuid_representation: Optional[int] = ...,
             unicode_decode_error_handler: Optional[str] = ...,
             tzinfo: Optional[datetime.tzinfo] = ...,
             type_registry: Optional[TypeRegistry] = ...,
             datetime_conversion: Optional[int] = ...,
-        ) -> "CodecOptions[_DocumentType]":
+        ) -> CodecOptions[_DocumentType]:
             ...
 
         # CodecOptions API
-        def with_options(self, **kwargs: Any) -> "CodecOptions[_DocumentType]":
+        def with_options(self, **kwargs: Any) -> CodecOptions[Any]:
             ...
 
         def _arguments_repr(self) -> str:
             ...
 
-        def _options_dict(self) -> Dict[Any, Any]:
+        def _options_dict(self) -> dict[Any, Any]:
```

Comment:
This should be kept as `Dict` to be consistent with the `pymongo` package typing.